src/DataTable/Pagination.tsx

```diff
@@ -288,6 +294,7 @@ export function Pagination({
                 )
               })
             : null}
+          {hasTrailingTruncation ? <TruncationStep key="truncation-1" /> : null}
```

Comment:
Previously these `TruncationSteps` had keys that aligned with the loop iteration, but I'm assuming the value here doesn't actually matter in this case.